nym-vpn-lib/src/platform/android.rs

```diff
@@ -288,3 +288,65 @@ fn is_public_ip(addr: IpAddr) -> bool {
         .iter()
         .any(|net| net.contains(addr))
 }
+
+#[derive(serde::Serialize)]
+enum AndroidControlMessageType {
+    // The VPN has been stopped successfully
+    StopSuccess,
+    // The VPN stopped, but with an error
+    StopError,
+    // The VPN sent an error, but has not stopped
+    Error,
+}
+
+#[derive(serde::Serialize)]
+struct AndroidControlMessage {
+    kind: AndroidControlMessageType,
+    message: Option<String>,
+}
+
+impl From<&NymVpnExitStatusMessage> for AndroidControlMessage {
+    fn from(status: &NymVpnExitStatusMessage) -> Self {
+        match status {
+            NymVpnExitStatusMessage::Stopped => AndroidControlMessage {
+                kind: AndroidControlMessageType::StopSuccess,
+                message: None,
+            },
+            NymVpnExitStatusMessage::Failed(error) => AndroidControlMessage {
+                kind: AndroidControlMessageType::StopError,
+                message: Some(error.to_string()),
+            },
+        }
+    }
+}
+
+impl From<&crate::Error> for AndroidControlMessage {
+    fn from(error: &crate::Error) -> Self {
+        match error {
+            crate::Error::StopError => AndroidControlMessage {
+                kind: AndroidControlMessageType::StopError,
+                message: Some(error.to_string()),
+            },
+            _ => AndroidControlMessage {
+                kind: AndroidControlMessageType::Error,
+                message: Some(error.to_string()),
+            },
+        }
+    }
+}
+
+fn send_android_control_message(exit_status_msg: &NymVpnExitStatusMessage) {
```

Comment:
`send` naming put my mind into looking for a different code here, until I realised it's actually indirectly "sent" because kotlin reads and interprets those changes. I think a better name would be `print` though

Reply:
ah yes good point